mirrord/operator/src/client.rs

```diff
@@ -45,29 +46,77 @@ pub enum OperatorApiError {
     Authentication(#[from] AuthenticationError),
     #[error("Can't start proccess because other locks exist on target")]
     ConcurrentStealAbort,
+    #[error("Invalid operator session - this is a bug, please report it. value: `{0}` err: `{1}`")]
+    InvalidOperatorSession(String, serde_json::Error),
 }
 
 type Result<T, E = OperatorApiError> = std::result::Result<T, E>;
 
 pub struct OperatorApi {
     client: Client,
-    features: Vec<OperatorFeatures>,
     target_api: Api<TargetCrd>,
     target_namespace: Option<String>,
     version_api: Api<MirrordOperatorCrd>,
     target_config: TargetConfig,
     on_concurrent_steal: ConcurrentSteal,
 }
 
+/// Data we store into environment variables for the child processes to use.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct OperatorSessionInformation {
+    pub session_id: String,
+    pub target: TargetCrd,
+    pub fingerprint: Option<String>,
+    pub operator_features: Vec<OperatorFeatures>,
+}
+
+impl OperatorSessionInformation {
+    pub fn new(
+        target: TargetCrd,
+        fingerprint: Option<String>,
+        operator_features: Vec<OperatorFeatures>,
+    ) -> Self {
+        Self {
+            session_id: rand::random::<u64>().to_string(),
+            target,
+            fingerprint,
+            operator_features,
+        }
+    }
+
+    /// Returns environment variable holding the information
+    pub const fn env_key() -> &'static str {
+        MIRRORD_OPERATOR_SESSION
+    }
+
+    /// Loads the information from environment variables
+    pub fn from_env() -> Result<Option<Self>> {
+        std::env::var(Self::env_key())
+            .ok()
+            .map(|val| {
+                serde_json::from_str(&val)
+                    .map_err(|e| OperatorApiError::InvalidOperatorSession(val, e))
+            })
+            .transpose()
+    }
+}
+
 impl OperatorApi {
-    pub async fn discover<P>(
+    /// Creates a new operator session, setting the session information in environment variables.
+    pub async fn create_session<P>(
```

Comment:
I'm not sure renaming it good because it can maybe create a session if operator exists otherwise it will return none, discover was fine IMO

Reply:
well we can split the API into three, but I think that attempting to create session if operator exists is ok